src/tests/test_algorithms.py

```diff
@@ -0,0 +1,52 @@
+from typing import Any, Dict
+
+import numpy as np
+import numpy.typing as npt
+import pytest
+from data_clusterer.algorithms import KMeansAlgorithm
+
+
+@pytest.fixture
+def kmeans_object():
+    return KMeansAlgorithm(n_clusters=3, random_state=42)
+
+
+@pytest.mark.parametrize(
+    "given_kwargs, expected_n_clusters, expected_random_state",
+    [
+        ({"n_clusters": 3, "random_state": 42}, 3, 42),
+        ({"n_clusters": None, "random_state": 50}, 8, 50),
+        ({"n_clusters": None, "random_state": None}, 8, None),
+    ],
+)
+def test_kmeans_init(
+    given_kwargs: Dict[str, int | None],
+    expected_n_clusters: int,
+    expected_random_state: int | None,
+):
+    test_instance = KMeansAlgorithm(**given_kwargs)
+    assert test_instance.model.n_clusters == expected_n_clusters
+    assert test_instance.model.random_state == expected_random_state
+
+
+def test_kmeans_fit_labels_length(
```

Comment:
we do not need to test scikit-learn function, just that it was called with proper parameters (write unit test with patching the scikit-learn method)